web/server/codechecker_server/api/mass_store_run.py

```diff
@@ -413,6 +413,51 @@ def __store_source_files(
 
         return file_path_to_id
 
+    def __add_blame_info(
+        self,
+        blame_root: str,
+        filename_to_hash: Dict[str, str]
+    ):
+        """
+        This function updates blame info in File and FileContent tables if
+        these were NULL. This function exists only because in earlier
+        CodeChecker versions blame info was not stored in these tables and
+        in a subsequent storage we can't update the tables for an unchanged
+        file since __store_source_files() updates only the ones that are in the
+        .zip file. This function stores blame info even if the corresponting
+        source file is not in the .zip file.
+        """
+        with DBSession(self.__Session) as session:
+            for subdir, _, files in os.walk(blame_root):
+                for f in files:
+                    blame_file = os.path.join(subdir, f)
+                    file_path = blame_file[len(blame_root.rstrip("/")):]
+                    blame_info, remote_url, tracking_branch = \
```

Comment:
We already getting and storing blame information on this part:
https://github.com/Ericsson/codechecker/blob/85dd52dc8f4d47fcf26419fcb44de1ff2a58924f/web/server/codechecker_server/api/mass_store_run.py#L384-L385

We have to process every git blame info file only once and not multiple times.

Reply:
I chose not adding blame info when inserting new `File` and `FileContent` fields because the alternative solution (i.e. returning the list of newly inserted fields) was not convenient. For example gathering the information whether a `File` is newly inserted would require to add a second return value for `add_file_record()` function and this info should be transferred to `__add_blame_info()` through further return values and function parameters.